### PR TITLE
chore(ci): Re-order soak results

### PR DESCRIPTION
**Pull request by jszwedko**
This:

* Adds `confidence` as a column to the detail table. Nathan mentioned
  this is easier to scan than the p-values and I agree.
* Removes the stderr columns from the summary table. I don't think this
  information is needed there. People generally just want to see the
  change in mean and confidence.

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

**Comment by fuchsnj** [comment]
>I found myself, when looking at the detail
table, typically wanting to see the most significant results and then
the magnitudes of their reported change.

I think the % change should stay as the sort column, but I'll see what this looks like. Mostly because there are usually some results with 100% confidence that are +- 0.01. You don't tend to get something that's a high percentage change but low confidence.

**Comment by blt** [comment]
> This:
> 
>     * Orders the summary and detail table by p-value followed by absolute
>       delta mean percentage. I found myself, when looking at the detail
>       table, typically wanting to see the most significant results and then
>       the magnitudes of their reported change.

I echo @fuchsnj and would prefer we keep the % change as the sort column. It's too bad Github markdown tables don't allow for sorting by clicking column headers. 

>     * Adds `confidence` as a column to the detail table. Nathan mentioned
>       this is easier to scan than the p-values and I agree.

Let's drop the p-value from the table. It's duplicated information and one is harder to read. 

>     * Removes the stderr columns from the summary table. I don't think this
>       information is needed there. People generally just want to see the
>       change in mean and confidence.

I'd prefer to keep stderr. Sans the ability to include graphs of the results its an important metric.

**Comment by jszwedko** [comment]
> > This:
> > ```
> > * Orders the summary and detail table by p-value followed by absolute
> >   delta mean percentage. I found myself, when looking at the detail
> >   table, typically wanting to see the most significant results and then
> >   the magnitudes of their reported change.
> > ```
> 
> I echo @fuchsnj and would prefer we keep the % change as the sort column. It's too bad Github markdown tables don't allow for sorting by clicking column headers.

The issue I've found is that often this will put statistically insignificant changes above significant ones. My attempt is to sort them in the order the reader is likely to care about them. If the change isn't statistically significant, I probably don't care about it.

> 
> > ```
> > * Adds `confidence` as a column to the detail table. Nathan mentioned
> >   this is easier to scan than the p-values and I agree.
> > ```
> 
> Let's drop the p-value from the table. It's duplicated information and one is harder to read.

👍 

> > ```
> > * Removes the stderr columns from the summary table. I don't think this
> >   information is needed there. People generally just want to see the
> >   change in mean and confidence.
> > ```
> 
> I'd prefer to keep stderr. Sans the ability to include graphs of the results its an important metric.

As a PR author, I'm not sure I've ever looked at that information. When would I take action on it? If we want to take action if there is a meaningful change in stderr, would that be covered by #10976 ? I did leave it in the detailed table.

**Comment by jszwedko** [comment]
For the sort order, consider https://github.com/vectordotdev/vector/pull/11365#issuecomment-1040322411 which places `http_to_http_acks` higher in the table than other soaks that have smaller deltas but are actually statistically significant.

Alternatively, we just sort the table alphabetically to remove an semblance that the order is important.

**Comment by blt** [comment]
> > > This:
> > > ```
> > > * Orders the summary and detail table by p-value followed by absolute
> > >   delta mean percentage. I found myself, when looking at the detail
> > >   table, typically wanting to see the most significant results and then
> > >   the magnitudes of their reported change.
> > > ```
> > 
> > 
> > I echo @fuchsnj and would prefer we keep the % change as the sort column. It's too bad Github markdown tables don't allow for sorting by clicking column headers.
> 
> The issue I've found is that often this will put statistically insignificant changes above significant ones. My attempt is to sort them in the order the reader is likely to care about them. If the change isn't statistically significant, I probably don't care about it.

IIRC it was @bruceg that originally suggested the sorting. I wonder if he has thoughts here. My preference is to retain sorting by %. The confidences matter less than the spread, at least to my way of reading the table. 

> > > ```
> > > * Adds `confidence` as a column to the detail table. Nathan mentioned
> > >   this is easier to scan than the p-values and I agree.
> > > ```
> > 
> > 
> > Let's drop the p-value from the table. It's duplicated information and one is harder to read.
> 
> +1
> 
> > > ```
> > > * Removes the stderr columns from the summary table. I don't think this
> > >   information is needed there. People generally just want to see the
> > >   change in mean and confidence.
> > > ```
> > 
> > 
> > I'd prefer to keep stderr. Sans the ability to include graphs of the results its an important metric.
> 
> As a PR author, I'm not sure I've ever looked at that information. When would I take action on it? If we want to take action if there is a meaningful change in stderr, would that be covered by #10976 ? I did leave it in the detailed table.

Might pole the team. If no one else uses the field I'm the oddball and we can drop it no problem.

**Comment by blt** [comment]
> For the sort order, consider [#11365 (comment)](https://github.com/vectordotdev/vector/pull/11365#issuecomment-1040322411) which places `http_to_http_acks` higher in the table than other soaks that have smaller % deltas but are actually statistically significant.
> 
> Alternatively, we just sort the table alphabetically to remove any semblance that the order is important.

Hmm, looking at the table above I'd almost want to see a _more_ complicated sort. Sort high/low on % change with soaks that share a similar confidence, say within 5% of one another. But, yeah, we're for sure at the limit of what you can scan from a table of numbers of this size. Have to choose something to highlight, give up.

**Comment by blt** [comment]
The confidence % _is_ much easier to scan than p-value.

**Comment by jszwedko** [comment]
> > For the sort order, consider [#11365 (comment)](https://github.com/vectordotdev/vector/pull/11365#issuecomment-1040322411) which places `http_to_http_acks` higher in the table than other soaks that have smaller % deltas but are actually statistically significant.
> > Alternatively, we just sort the table alphabetically to remove any semblance that the order is important.
> 
> Hmm, looking at the table above I'd almost want to see a _more_ complicated sort. Sort high/low on % change with soaks that share a similar confidence, say within 5% of one another. Or above threshold, below threshold. But, yeah, we're for sure at the limit of what you can scan from a table of numbers of this size. Have to choose something to highlight, give up.

Yeah, I was thinking about that too. I almost want two separate tables: one for significant, one for not, but then we are also explicitly saying changes less than a threshold are not significant. Taking a step back  I've been looking at the detail table frequently to survey change because the threshold seemed to be masking many of them and some of the soaks we are working on improving are marked as erratic and so also will never appear in the summary table. I'm almost inclined to not bother sorting the detail table by any of the metrics given there doesn't seem to be a great set to do so along. A dynamic table with sortable headers would be 💯

**Comment by jszwedko** [comment]
@blt @fuchsnj I replaced the existing sort order since the change seems to be contentious. Would you want to give this another 👀 ?

File: soaks/bin/analyze_experiment [hunk 1]
```diff
@@ -127,16 +128,14 @@ p_value_violation = results['p-value'] < args.p_value
 erratic_violation = results.erratic == True
 
 changes = results[p_value_violation & drift_filter & ~declared_erratic].copy(deep=True)
-changes['Δ confidence'] = changes['p-value'].apply(common.confidence)
```
Comment (fuchsnj) [inline soaks/bin/analyze_experiment:130]: I thought we still wanted the new condifence column instead of p-value?

Comment (fuchsnj) [reply soaks/bin/analyze_experiment:130]: eh, ignore this. I read it wrong

**Review by fuchsnj — APPROVED** [review]

File: soaks/bin/analyze_experiment (hunk 1 above)
Comment (jszwedko) [reply soaks/bin/analyze_experiment:130]: Ah, yeah, I moved it up so it appears in both tables.

**Pull request merged**